### PR TITLE
chore: Version 1.0.0

### PR DESCRIPTION
**Pull request by miquelbeltran**
### Description :memo:
- **Purpose**: Merge and publish v1.0.0
- **Approach**: 

**Type of change**

- Release

**Updates**

Changelog:

- feat: #219 Better Send parameters (#221) (2024-05-17)
- feat: Breadcrumbs (#210) (2024-05-15)
- fix: #138 New APM Bridge Setup (#222) (2024-05-20)
- docs: Update status badge in README.md (#209) (2024-05-10)
- refactor: #205 Setup eslint style (#207) (2024-05-09)
- refactor: #197 Refactor to use Promises internally (#200) (2024-05-09)
- refactor: #184 Cleanup debug/log messages and styles (#194) (2024-05-08)
- chore(deps-dev): bump typescript-eslint from 7.8.0 to 7.9.0 (#229) (2024-05-20)
- chore(deps-dev): bump @eslint/js from 9.2.0 to 9.3.0 (#228) (2024-05-20)
- chore(deps-dev): bump @types/node from 20.12.11 to 20.12.12 (#227) (2024-05-20)
- chore(deps-dev): bump @stylistic/eslint-plugin from 2.0.0 to 2.1.0 (#214) (2024-05-13)
- chore(deps-dev): bump tap from 18.7.2 to 18.8.0 (#213) (2024-05-13)
- chore(deps-dev): bump @types/node from 20.12.8 to 20.12.11 (#212) (2024-05-13)
- chore(deps-dev): bump semver from 7.6.0 to 7.6.2 (#211) (2024-05-13)
- chore: code cleanup (#225) (2024-05-20)
- chore: apply prettier to all files in examples (#226) (2024-05-19)

### Test plan :test_tube:

- Tested with express example

### Author to check :eyeglasses:
- [x] Project and all contained modules builds successfully
- [x] Self-/dev-tested 
- [x] Unit/UI/Automation/Integration tests provided where applicable
- [x] Code is written to standards
- [x] Appropriate documentation written (code comments, internal docs)

### Reviewer to check :heavy_check_mark:
- [x] Project and all contained modules builds successfully
- [x] Change has been dev-/reviewer-tested, where possible
- [x] Unit/UI/Automation/Integration tests provided where applicable
- [x] Code is written to standards
- [x] Appropriate documentation written (code comments, internal docs)

### CODE REVIEW

**Review by sumitramanga — APPROVED** [review]
I've tested and code reviewed the majority! LGTM, nice work 🚀

**Comment by miquelbeltran** [comment]
@sumitramanga should I wait for more reviews to merge and publish?

**Comment by sumitramanga** [comment]
Waiting on Panos to review first ! @miquelbeltran

File: README.md
Comment (PanosNB) [inline README.md:493]: Add a full example of how this works. Something like:
`
require('raygun-apm/http');
const raygun = require('raygun');
raygun.setApmBridge();
`

File: CHANGELOG.md
Comment (PanosNB) [inline CHANGELOG.md:23]: What were we doing before and as above, can we support a deprecated version?

File: README.md
Comment (PanosNB) [inline README.md:496]: That looks like a nasty setup. Can we make it without arguments and infer it? Still have the explicit call with the argument as fallback

File: examples/apm-sample/README.md
Comment (PanosNB) [inline examples/apm-sample/README.md:17]: A customer yesterday was asking if we support `pnpm` instead. Investigate and please add to the docs as an alternative

File: examples/apm-sample/app.js
Comment (PanosNB) [inline examples/apm-sample/app.js:3]: Should we be encouraging no API keys in code to set a good standard? Could be via .env

File: examples/express-sample/bin/www
Comment (PanosNB) [inline examples/express-sample/bin/www:15]: What if the user does not use `process.env.PORT` to set up Express? Should we be pulling this elsewhere, if possible?

**Review by PanosNB — CHANGES_REQUESTED** [review]
My main concern is introducing breaking changes. I'd like us to consider adding deprecated, backwards-compatible functions if possible. Let's discuss to reach a consensus on this :-)

Apart from that, great work!! I can see a lot was done!

File: examples/apm-sample/README.md
Comment (sumitramanga) [reply examples/apm-sample/README.md:17]: Let's make this an issue to resolved outside of this release

File: examples/apm-sample/app.js
Comment (sumitramanga) [reply examples/apm-sample/app.js:3]: Because this is an example app, I would think this is not necessary to add in

File: examples/express-sample/bin/www
Comment (sumitramanga) [reply examples/express-sample/bin/www:15]: Because this is an example app, we should not have to worry about this? 🤔 We are using the standard way to set up ports I would think. Happy to be challenged! Such as, what other way would people use commonly :)

File: CHANGELOG.md
Comment (sumitramanga) [reply CHANGELOG.md:20]: If it makes sense to add in the deprecation, we could do this now whilst we discuss the final decision. It looks like we've done something in the past about deprecating (which we may need to remove - https://github.com/MindscapeHQ/raygun4node/blob/8e37c10086ac6d7e7d5236831e50a377d4ef9f49/lib/raygun.ts#L159)

Comment (miquelbeltran) [reply CHANGELOG.md:20]: We provide a backwards compatible method, is `sendWithCallback()`.

Comment (miquelbeltran) [reply CHANGELOG.md:20]: > it could as simple as `function send(error, customData={}, request,={} tags={}){ return send(error, {customData: customData, request: request, tags: tags});}`

JavaScript doesn't support overloading methods. There ware some workarounds but have limitations.

File: README.md
Comment (miquelbeltran) [reply README.md:493]: It is already in `examples/apm-sample/app.js`

File: README.md
Comment (miquelbeltran) [reply README.md:496]: Can you take a look at the PR where we introduced this and review it? https://github.com/MindscapeHQ/raygun4node/pull/222

File: README.md
Comment (miquelbeltran) [reply README.md:493]: Ah you mean in the README, got it.

File: examples/apm-sample/app.js
Comment (miquelbeltran) [reply examples/apm-sample/app.js:3]: `raygun-apm` gets the api key from env variable, which we should probably do as well eventually, out of scope for this release IMO.

File: examples/express-sample/bin/www
Comment (miquelbeltran) [reply examples/express-sample/bin/www:15]: `process.env.PORT || "3000"` works that if the user doesn't have a `$PORT` in their env vars, `3000` will be used by default.

Yes, agreed with @sumitramanga I guess this is standard express:

image

**Comment by miquelbeltran** [comment]
@PanosNB I will do a new PR with some of the suggestions.

I've also replied on Slack regarding the discussion on breaking changes in the `send()` method.

Please take a look at my other replies in your review, thanks!

File: CHANGELOG.md
Comment (miquelbeltran) [reply CHANGELOG.md:23]: The previous setup process was causing compilation errors to some users and was not possible to properly unit test it, it is explained in detail here: https://github.com/MindscapeHQ/raygun4node/pull/222

File: README.md
Comment (miquelbeltran) [reply README.md:334]: I will add more info.

File: README.md
Comment (miquelbeltran) [reply README.md:154]: I will fix this.

File: README.md
Comment (miquelbeltran) [reply README.md:493]: But I think the current text is correct. In other parts of the readme, we don't reference to the `require` for raygun client, and also the `require('raygun-apm/http')` is not required to set up the APM bridge. I also link to the official APM setup process.

**Comment by miquelbeltran** [comment]
APM Bridge changes reverted

**Review by PanosNB — APPROVED** [review]
Thanks Miguel, great work!

**Pull request merged**
